azure-devops/checkout-sources.yml

```
@@ -43,9 +43,10 @@ steps:
     )
 
     git fetch --filter=tree:0 --depth=1 llvm $(${{ parameters.llvmSHAVar }})
-    git reset --quiet FETCH_HEAD
     git sparse-checkout init --cone
     git sparse-checkout set libcxx/test libcxx/utils/libcxx llvm/utils/lit
+    git reset --hard FETCH_HEAD

```

We only over care about libcxx so maybe we could add a -- libcxx/ to tell git that?

---

According to my understanding, the previous line's sparse-checkout patterns have already done that. Only those subdirectories (recursively) and the files directly in the root folder exist after this reset.

---

I've found that if a large number of files need to get reset for whatever reason passing `--quiet` significantly speeds up the command.

```suggestion
    git reset --quiet --hard FETCH_HEAD
```

---

Thanks - I will add that, and also to the `git clean` which supports `--quiet`.